pkg/reconciler/common/targetnamespace.go

```diff
@@ -18,33 +18,177 @@ package common
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	"github.com/tektoncd/operator/pkg/reconciler/shared/hash"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/logging"
 )
 
-func CreateTargetNamespace(ctx context.Context, labels map[string]string, obj v1alpha1.TektonComponent, kubeClientSet kubernetes.Interface) error {
-	ownerRef := *metav1.NewControllerRef(obj, obj.GroupVersionKind())
-	namespace := &corev1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: obj.GetSpec().GetTargetNamespace(),
-			Labels: map[string]string{
-				"operator.tekton.dev/targetNamespace": "true",
-			},
-			OwnerReferences: []metav1.OwnerReference{ownerRef},
-		},
+const (
+	labelKeyTargetNamespace     = "operator.tekton.dev/targetNamespace"
+	labelKeyOpenshiftMonitoring = "openshift.io/cluster-monitoring"
+)
+
+func ReconcileTargetNamespace(ctx context.Context, tektonComponent v1alpha1.TektonComponent, kubeClientSet kubernetes.Interface) error {
+	// get logger
+	logger := logging.FromContext(ctx)
+
+	logger.Debugw("reconciling target namespace",
+		"targetNamespace", tektonComponent.GetSpec().GetTargetNamespace(),
+	)
+
+	// owner reference used for target namespace
+	ownerRef := *metav1.NewControllerRef(tektonComponent, tektonComponent.GroupVersionKind())
```

Comment:
This should be moved below for readability just above the block where it will be used for better readability.